cmd/broker/ingress/main.go

```diff
@@ -70,26 +76,45 @@ func main() {
 	}
 
 	err = mgr.Add(&runnableServer{
-		logger: logger,
-		s:      s,
+		logger:          logger,
+		s:               s,
+		ShutdownTimeout: writeTimeout,
 	})
 	if err != nil {
 		logger.Fatal("Unable to add runnableServer", zap.Error(err))
 	}
 
+	// Metrics
+	e, err := prometheus.NewExporter(prometheus.Options{Namespace: componentName})
```

Comment:
If we have multiple brokers in a namespace or even different brokers in different namespaces, will all be sending metrics as if they were the same component? I'm not aware of exporters, but I guess I got worried about that componentName to be the same for everyone. In other words, would it be easy to distinguish from which broker ingress the metrics came from? Does it make sense?

Reply:
Good point, we should probably add a tag for Broker name. I think the easiest way to do this is to add the Broker name to the Pod env vars.

The Prometheus server can also apply tags server-side, but I don't know if namespace is one of those.